webui-src/app/people/people_ownids.js

```diff
@@ -1,213 +1,379 @@
       ),
       m(
-        'button',
+        "button",
         {
-          onclick: () =>
-            rs.rsJsonApiRequest(
-              '/rsIdentity/createIdentity',
-              {
-                name,
-                pseudonimous,
-                pgpPassword,
-              },
-              data => {
-                const message = data.retval
-                  ? 'Successfully created identity.'
-                  : 'An error occured while creating identity.';
-                widget.popupMessage([
-                  m('h3', 'Create new Identity'),
-                  m('hr'),
-                  message,
-                ]);
-                m.redraw();
-              },
-            ),
+          onclick: () => {
+            if (pseudonimous == false) {
+              let paraphase = "";
+              widget.popupMessage([
+                m("i.fas.fa-user-edit"),
+                m("h3", "Enter your passpharse"),
+                m("hr"),
+
+                m("input[type=password][placeholder=Passpharse]", {
+                  style: "margin-top:50px;width:80%",
+                  oninput: (e) => {
+                    paraphase = e.target.value;
+                  },
+                }),
+                m(
+                  "button",
+                  {
+                    style: "margin-top:160px;",
+                    onclick: () => {
+                      rs.rsJsonApiRequest(
+                        "/rsIdentity/getOwnSignedIds",
+                        {},
+                        (owns) => {
+                          if (owns.ids.length > 0) {
+                            console.log(owns.ids);
+                            id = owns.ids[0];
+                            console.log(id);
+                          }
+                          rs.rsJsonApiRequest(
+                            "/rsIdentity/createIdentity",
+                            {
+                              id: id,
+                              name: name,
+                              //avatar: v.attrs.details.mAvatar.mData.base64,
+                              pseudonimous: pseudonimous,
+                              pgpPassword: paraphase,
+                            },
+                            (data) => {
+                              const message = data.retval
+                                ? "Successfully created identity."
+                                : "An error occured while creating identity.";
+                              widget.popupMessage([
+                                m("h3", "Create new Identity"),
+                                m("hr"),
+                                message,
+                              ]);
+                            }
+                          );
+                        }
+                      );
+                    },
+                  },
+                  "Enter"
+                ),
+              ]);
+            } else
+              rs.rsJsonApiRequest(
```

Comment:
The `id` param is the only difference between signed and unsigned, so we can probably create a single popup component for both and customize with props.